packages/vuetify/src/components/VTreeview/VTreeview.ts

```diff
@@ -303,6 +307,121 @@ export default mixins(
+        }
+        this.updateFocus(nextNode, true)
+      // is at the end of an group index
+      } else if (currentIndex === parentSiblings.length - 1) {
+        let nextNode = key
+        // if node is open goto first child
+        if (node.isOpen && node.children.length > 0) {
+          nextNode = node.children[0]
+        } else {
+          // locate the top most parent of a node
+          const topParents = this.items.map((c: any) => getObjectValueByPath(c, this.itemKey))
+          const getNextTopNode = function (searchKey: string | number,
+            nodes: Record<string | number, NodeState>, parents: string|number[]): string | number {
+            const n = nodes[searchKey]
+            if (n.parent) {
+              return getNextTopNode(n.parent, nodes, parents)
+            } else {
+              return searchKey
+            }
+          }
+          const topParent = getNextTopNode(key, this.nodes, topParents)
+          const nextNodeIndex = topParents.indexOf(topParent)
+          // if top mode parent has an index to go down 1
+          if (nextNodeIndex > -1 && nextNodeIndex < topParents.length - 1) {
+            nextNode = topParents[nextNodeIndex + 1]
+          }
+        }
+        this.updateFocus(nextNode, true)
+      }
+    },
+    keyMoveLeft (key: string | number): void {
+      if (!key || !this.nodes.hasOwnProperty(key)) {
+        return
+      }
+
+      const node = this.nodes[key]
+      // closes current node
+      if (node.isOpen) {
+        this.updateOpen(key, false)
+        this.updateFocus(key, true)
+      // if parent, focus on parent
+      } else if (node.parent) {
+        this.updateFocus(node.parent, true)
+      } else {
+        if (this.focusCache !== key) {
+          this.updateFocus(key, true)
+        }
+      }
+    },
+    keyMoveRight (key: string | number): void {
+      if (!key || !this.nodes.hasOwnProperty(key)) {
+        return
+      }
+      const node = this.nodes[key]
+      if (node.isOpen && node.children.length > 0) {
+        this.updateFocus(node.children[0], true)
+      } else {
+        // reset focus to current index if needed
+        if (this.focusCache !== key) {
+          this.updateFocus(key, true)
+        }
+      }
+    },
+    keyMoveUp (key: string | number): void {
+      if (!key || !this.nodes.hasOwnProperty(key)) {
+        return
+      }
+
+      const node = this.nodes[key]
+      const parentSiblings = node.parent ? this.nodes[node.parent].children
+        : this.items.map((c: any) => getObjectValueByPath(c, this.itemKey))
+      const currentIndex = parentSiblings.indexOf(key)
+      // is there an index to go up 1
+      if (currentIndex > 0) {
+        let previousNode = parentSiblings[currentIndex - 1]
+        // locate a previous parentNode that is open and has children, goto last child
+        const getLastOpenedChildNode = (searchKey: string | number, nodes: Record<string | number, NodeState>): string | number => {
```

Comment:
This function is redundant, it's just being called straight away then discarded.